code/modules/client/preferences.dm

```diff
@@ -1819,6 +1822,11 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 					if(!pickedbar)
 						return
 					bar_choice = pickedbar
+				if ("preferred_ai_module")
```

Comment:
Rework this to be like other donator stuff in process_link, as it stands this is open to href exploits